Datetime.strptime error in wxPython why

```
# coding=utf-8
import wx
import sys
import datetime

if __name__ == "__main__":
    print("wxpython version:", wx.version())
    print("Python version:", sys.version)
    print("Run before wx:", datetime.datetime.strptime("02/02/2020", '%d/%m/%Y'), "normal")
    app = wx.App()
    print("Run after wx:", datetime.datetime.strptime("02/02/2020", '%d/%m/%Y'), "error")

    # frame = MainFrame()
    # frame.SetSize(0, 0, 1200, 750)
    # frame.Center()
    # frame.Show()
    app.MainLoop()
```


```
wxpython version: 4.1.0 msw (phoenix) wxWidgets 3.1.4
Python version: 3.6.8 (tags/v3.6.8:3c6b436a57, Dec 24 2018, 00:16:47) [MSC v.1916 64 bit (AMD64)]
Run before wx: 2020-02-02 00:00:00 normal
Traceback (most recent call last):
  File "K:/test/a.py", line 14, in <module>
    print("Run after wx:", datetime.datetime.strptime("02/02/2020", '%d/%m/%Y'), "error")
  File "C:\Program Files\Python36\lib\_strptime.py", line 565, in _strptime_datetime
    tt, fraction = _strptime(data_string, format)
  File "C:\Program Files\Python36\lib\_strptime.py", line 334, in _strptime
    if (_getlang() != locale_time.lang or
  File "C:\Program Files\Python36\lib\_strptime.py", line 31, in _getlang
    return locale.getlocale(locale.LC_TIME)
  File "C:\Program Files\Python36\lib\locale.py", line 581, in getlocale
    return _parse_localename(localename)
  File "C:\Program Files\Python36\lib\locale.py", line 490, in _parse_localename
    raise ValueError('unknown locale: %s' % localename)
ValueError: unknown locale: zh-CN
```

I'm seeing this too with wxPython 4.1.0 (and Python 3.8.1 MSW 64 bit, for the record).
The same traceback occurs even if the locale is set explicitly through wx:
```
import wx
import datetime
app = wx.App()
loc = wx.Locale(wx.LANGUAGE_ENGLISH_UK)
print(datetime.datetime.strptime('2020', "%Y"))
```
However, if the locale is set using the Python locale module, it doesn't:
```
import wx
import datetime, locale
app = wx.App()
locale.setlocale(locale.LC_ALL, 'en_GB')
print(datetime.datetime.strptime('2020', "%Y"))
```

No traceback in this case. It seems to be a usable workaround. 

Note that some older 4.0.x wxPython's insisted on the opposite, requiring wx.Locale instead of Python locale, as seen in this assertion failure from 4.0.4 wx.Image.ConvertToBitmap: 
`
wx._core.wxAssertionError: C++ assertion "strcmp(setlocale(LC_ALL, NULL), "C") == 0" failed at ..\..\src\common\intl.cpp(1579) in wxLocale::GetInfo(): You probably called setlocale() directly instead of using wxLocale and now there is a mismatch between C/C++ and Windows locale.
`

Tried the `InitLocale` override, and yes, it works to set the "C" locale.  You can't subsequently set a different locale; any other locale than the C locale (specifically, `wx.LANGUAGE_ENGLISH`) or the default locale results in this error message from `wx.Locale`: 

`wx._core.wxAssertionError: C++ assertion ""!szLocale.empty()"" failed at ..\..\src\common\intl.cpp(304) in wxLocale::Init(): no locale to set in wxLocale::Init()`

--- a/.github/issue_template.md
+++ b/.github/issue_template.md
@@ -11,8 +11,12 @@
 
 **Description of the problem**:
 
-<!-- if possible please include a small runnable application that demonstrates the problem -->
+<!-- if possible please include a small runnable application in the details section below that demonstrates the problem -->
+
+<details>
+  <summary>Code Example (click to expand)</summary>
 
 ```python
 # Put code sample here
 ```
+</details>

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -28,6 +28,13 @@ New and improved in this release:
   Python environment used for the build. A dependency has been added to
   requirements/devel.txt to help ensure that the correct version is installed.
 
+* Change wx.App.InitLocale to just do `locale.setlocale(locale.LC_ALL, "C")`
+  to undo what Python (3.8+ on Windows) does. This lets wxWidgets start with an
+  uninitialized locale as it expects. (#1637)
+
+* Fixed issues related to `time_t` being treated as a 32-bit value on Windows.
+  (#1910)
+
 
 
 4.1.1 "An attitude of gratitude"

--- a/demo/Timer.py
+++ b/demo/Timer.py
@@ -219,6 +219,12 @@ class TestPanel(sp.ScrolledPanel):
         self.log.write("got wx.PyTimer event\n")
 
 
+    def ShutdownDemo(self):
+        for t in ['t1', 't2', 't3', 't4']:
+            if hasattr(self, t):
+                print('Stopping:', t)
+                timer = getattr(self, t)
+                timer.Stop()
 
 #----------------------------------------------------------------------
 

--- a/etg/app.py
+++ b/etg/app.py
@@ -521,33 +521,28 @@ def run():
 
             PyFunctionDef('InitLocale', '(self)',
                 doc="""\
-                    Try to ensure that the C and Python locale is in sync with the wxWidgets
-                    locale on Windows. If you have troubles from the default behavior of this
-                    method you can override it in a derived class to behave differently.
-                    Please report the problem you encountered.
+                    Starting with version 3.8 on Windows, Python is now setting the locale
+                    to what is defined by the system as the default locale. This causes
+                    problems with wxWidgets which expects to be able to manage the locale
+                    via the wx.Locale class, so the locale will be reset here to be the
+                    default "C" locale settings.
+
+                    If you have troubles from the default behavior of this method you can
+                    override it in a derived class to behave differently. Please report
+                    the problem you encountered.
                     """,
                 body="""\
-                    self.ResetLocale()
-                    if 'wxMSW' in PlatformInfo:
+                    if _sys.platform.startswith('win') and _sys.version_info > (3,8):
                         import locale
-                        try:
-                            lang, enc = locale.getdefaultlocale()
-                            self._initial_locale = wx.Locale(lang, lang[:2], lang)
-                            locale.setlocale(locale.LC_ALL, lang)
-                        except (ValueError, locale.Error) as ex:
-                            target = wx.LogStderr()
-                            orig = wx.Log.SetActiveTarget(target)
-                            wx.LogError("Unable to set default locale: '{}'".format(ex))
-                            wx.Log.SetActiveTarget(orig)
+                        locale.setlocale(locale.LC_ALL, "C")
                     """),
 
             PyFunctionDef('ResetLocale', '(self)',
                 doc="""\
-                    Release the wx.Locale object created in :meth:`InitLocale`.
-                    This should reset the application's locale to the previous settings.
+                    This method is now a NOP and will be deprecated.
                     """,
                 body="""\
-                    self._initial_locale = None
+                    pass
                     """),
 
             PyFunctionDef('Get', '()', isStatic=True,

--- a/etg/defs.py
+++ b/etg/defs.py
@@ -73,7 +73,7 @@ def run():
     td = module.find('wxUIntPtr')
     module.insertItemAfter(td, etgtools.TypedefDef(type='wchar_t', name='wxUChar'))
     module.insertItemAfter(td, etgtools.TypedefDef(type='wchar_t', name='wxChar'))
-    module.insertItemAfter(td, etgtools.TypedefDef(type='long', name='time_t'))
+    module.insertItemAfter(td, etgtools.TypedefDef(type='wxInt64', name='time_t'))
     module.insertItemAfter(td, etgtools.TypedefDef(type='long long', name='wxFileOffset'))
     module.insertItemAfter(td, etgtools.TypedefDef(type='SIP_SSIZE_T', name='ssize_t'))
     module.insertItemAfter(td, etgtools.TypedefDef(type='unsigned char', name='byte', pyInt=True))

--- a/etg/intl.py
+++ b/etg/intl.py
@@ -61,6 +61,8 @@ def run():
     import os
     _localedir = os.path.join(os.path.dirname(__file__), "locale")
     if os.path.exists(_localedir):
+        if isinstance(_localedir, (bytes, bytearray)):
+            _localedir = _localedir.decode(_sys.getfilesystemencoding())
         Locale.AddCatalogLookupPathPrefix(_localedir)
     del os
     #----------------------------------------------------------------------------

--- a/wx/lib/agw/aui/framemanager.py
+++ b/wx/lib/agw/aui/framemanager.py
@@ -3761,7 +3761,7 @@ def GetNotebookRoot(panes, notebook_id):
 
 def EscapeDelimiters(s):
     """
-    Changes ``;`` into ``\`` and ``|`` into ``\|`` in the input string.
+    Changes ``;`` into ``\\`` and ``|`` into ``|\\`` in the input string.
 
     :param string `s`: the string to be analyzed.
 

